.github/workflows/cf-lambda-shared-deploy.yml

```diff
@@ -0,0 +1,105 @@
+# Shared workflow for publishing Lambda@Edge functions to
+# CloudFront distributions
+name: Zip Lambda and Redeploy Distribution
+
+on:
+  workflow_call:
+    inputs:
+      AWS_REGION:
+        required: true
+        type: string
+      GHA_ROLE:
+        required: true
+        type: string
+      ENVIRONMENT:
+        required: true
+        type: string
+      UPLOAD_ZIP:
+        required: false
+        type: boolean
+        default: false
+      ZIP_FILE_NAME:
+        required: false
+        type: string
+        default: some_file_name
+      TF_WORKSPACE:
+        required: true
+        type: string
+      TF_AUTO_APPLY:
+        required: false
+        type: boolean
+        default: false
+
+# Set defaults
+defaults:
+  run:
+    shell: bash
+
+## Add shared Environment Variables across jobs here ##
+env:
+  TF_CLOUD_ORGANIZATION: "${{ vars.TF_CLOUD_ORGANIZATION }}"
+  TF_API_TOKEN: "${{ secrets.TF_API_TOKEN }}"
+
+jobs:
+  upload-zip:
+    # Only run if the caller workflow set UPLOAD_ZIP to `true` and the call is NOT from dependabot
+    if: ${{ inputs.UPLOAD_ZIP && github.triggering_actor != 'dependabot[bot]' }}
+    name: Zip code and upload to S3
+    runs-on: ubuntu-latest
+    # This line adds a check for the user which is requesting the PR. As long as its not dependabot, we go ahead and run it. 
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      # First we authenticate to AWS account, based on the ENVIRONMENT input
+      - uses: actions/checkout@v3
+      - name: DEV Configure AWS credentials
+        # Only run this step if the environment is "dev"
+        if: ${{ inputs.ENVIRONMENT == 'dev' }}
+        uses: aws-actions/configure-aws-credentials@v2
```

Comment:
@v3 is available for [configure-aws-credentials](https://github.com/aws-actions/configure-aws-credentials/releases) (actually v4 is available but i haven't tested that one yet) 
might be worth updating and testing one more time before this goes into production

Reply:
ha ha! there are a few actions version updates that I need to review, but I don't think that right now is the time (and they are all seemingly logged w/ dependabot PRs, so I won't lose track of them)